### ISSUE TITLE
macvtap interface dosen't get created

### ISSUE THREAD

**Issue by Kabbone** [astro/microvm.nix#151]
I tried to create a microvm with macvtap interface fully declarative. The problem is:

```
➜ systemctl status -l microvm-macvtap-interfaces@gitea.service
○ microvm-macvtap-interfaces@gitea.service - Setup MicroVM 'gitea' MACVTAP interfaces
     Loaded: loaded (/etc/systemd/system/microvm-macvtap-interfaces@.service; static)
     Active: inactive (dead)
  Condition: start condition unmet at Sat 2023-10-14 16:02:11 CEST; 1min 14s ago
             └─ ConditionPathExists=/var/lib/microvms/gitea/current/share/microvm/macvtap-interfaces was not met

Okt 14 15:59:55 nuts systemd[1]: Setup MicroVM 'gitea' MACVTAP interfaces was skipped because of an unmet condition check (ConditionPathExists=/var/lib/microvms/gitea/current/share/microvm/macvtap-interfaces).
```

So the macvtap-interfaces file doesn't get created.

microvm.nix:
```
{ microvm, nixpkgs, ... }:
let
  name = "test";
in
{
    vms = {
      ${name} = {
        pkgs = import nixpkgs { system = "x86_64-linux"; };

        config = {
          networking.hostName = "${name}-runner";
          users.users.root.password = "";
          microvm = {
            hypervisor = "cloud-hypervisor";
            vcpu = 4;
            mem = 4096;
            interfaces = [
            {
              type = "macvtap";
              id = "${name}-tap";
              mac = "02:00:00:00:00:01";
              macvtap = {
                  link = "enp34s0";
                  mode = "bridge";
              };
            } ];
             shares = [{
              source = "/nix/store";
              mountPoint = "/nix/.ro-store";
              tag = "ro-store";
              proto = "virtiofs";
            }];
          };
        system.stateVersion = "23.05";
        };
      };
    };
  };
}
```

I'm still kind of a novice on nix, but the setup looks to me basically similar like in #106

**Comment by astro** [comment]
Thank you for reporting!

The bug was introduced in c1c71f28c5179d5d2a95d9042d376c75c79e98de. Fixed in 38ce07ca9c5c02c391682defe0bfff947cc5d7a1.

**Comment by Kabbone** [comment]
Thx @astro, insane fast fix, works now :+1:
